csharp-advanced-topics/VolatileInterlockedLockInCsharp/VolatileInterlockedLockInCsharp/Account.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace VolatileInterlockedLockInCsharp
+{
+    public class Account
+    {
+        private int _balance;
+        private volatile int _balanceVolatile;
+        private int _balanceLock;
+        private int _balanceInterlocked;
+        private object _lock = new object();
+
+        public int Balance
+        {
+            get => _balance;
+            set => _balance = value;
+        }
+
+        public int BalanceVolatile
+        {
+            get => _balanceVolatile;
+            set => _balanceVolatile = value;
+        }
+
+        public int BalanceLock
+        {
+            get
+            {
+                lock (_lock)
+                {
+                    return _balanceLock;
+                }
+            }
+            set
+            {
+                lock (_lock)
+                {
+                    _balanceLock = value;
+                }
+            }
+        }
+
+        public int BalanceInterlocked
+        {
+            get => _balanceInterlocked;
+            set => _balanceInterlocked = value;
```

Comment:
For consistency with the other properties and code, we should be using the Interlocke statements here. We are using lock in the BalanceLock case, so we should "mirror" that style of behavior here.
So we would use `CompareExchange()` for the get and `Exchange()` for the set.

Reply:
Done